app/routes/project/index.js

```diff
@@ -10,15 +10,13 @@ export default Route.extend({
   userSubscriptions: service(),
 
   model() {
-    let project = this.modelFor('project');
-
-    return RSVP.hash({
-      project,
-      subscription: this.get('userSubscriptions').fetchForProject(project)
+    return this.modelFor('project').reload().then((project) => {
```

Comment:
I took another look at this, and tried to rewrite it as a regular `queryRecord`. I also made use of restructuring, to make it a bit cleaner.

Here's what it looks like now:

```Javascript
model() {
  return this.modelFor('project').reload().then((project) => {
    let subscription = this.get('userSubscriptions').fetchForProject(project);
    return RSVP.hash({ project, subscription });
  });
}
```

Here's what it looks like as a regular hook

```Javascript
model(params) {
  return this.store.queryRecord('project', { 
    slug: params.project_slug, 
    sluggedRouteSlug: params.slugged_route_slug 
  }, { reload: true }).then((project) => {
    let subscription = this.get('userSubscriptions').fetchForProject(project);
    return RSVP.hash({ project, subscription });
  });
},
```

Here' what it looks like with destructuring:


```Javascript
model({ project_slug: slug, slugged_route_slug: sluggedRouteSlug}) {
  return this.store.queryRecord('project', { slug, sluggedRouteSlug }, { reload: true }).then((project) => {
    let subscription = this.get('userSubscriptions').fetchForProject(project);
    return RSVP.hash({ project, subscription });
  });
},
```

The first approach still feels like an anti-pattern to me, but it is also drier as is and the only truly clean solution would be something along the lines of server sent events anyway.

Really, we may be giving too much importance to reloading. In my opinion, the only case where we should stress about reloading the project is when the user has donated. In that specific case, they are actually expecting for donation progress to change.

In cases where a project admin has changed something about a project, like a description, it should not be that important for the regular user to see that change immediately. I think having that happen somewhere down the line, when the page refreshes or something like that, ought to be good enough, until we have server sent events, at least.

Either way, for me, it would be enough to just stick to explicit reloading when making a donation. The current implementation, where we explicitly reload on each subroute change seems like overkill to me, but I would be ok with keeping it. I would keep it as is, because the preferable solution would be server sent events, so it's drier and easier to change this way.